### ISSUE TITLE
Error when attempting edit

### ISSUE THREAD

**Issue by baailey** [Deltik/SignEdit#33]
![image](https://user-images.githubusercontent.com/62433969/229339971-653bb645-befc-4e03-96c4-9a444bd28354.png)

```
02.04 17:52:02 [Server] INFO sxjx issued server command: /editsign 3 &d&lAutmn Egg
02.04 17:52:02 [Server] ERROR java.lang.ClassCastException: class org.bukkit.craftbukkit.v1_19_R2.block.data.CraftBlockData cannot be cast to class org.bukkit.block.data.type.WallSign (org.bukkit.craftbukkit.v1_19_R2.block.data.CraftBlockData and org.bukkit.block.data.type.WallSign are in unnamed module of loader java.net.URLClassLoader @5c29bfd)
02.04 17:52:02 [Server] INFO at SignEdit.jar//net.deltik.mc.signedit.integrations.BreakReplaceSignEditValidator.getBlockAgainstWallSign(BreakReplaceSignEditValidator.java:94)
02.04 17:52:02 [Server] INFO at SignEdit.jar//net.deltik.mc.signedit.integrations.BreakReplaceSignEditValidator.getBlockAgainst(BreakReplaceSignEditValidator.java:86)
02.04 17:52:02 [Server] INFO at SignEdit.jar//net.deltik.mc.signedit.integrations.BreakReplaceSignEditValidator.validateBlockPlace(BreakReplaceSignEditValidator.java:71)
02.04 17:52:02 [Server] INFO at SignEdit.jar//net.deltik.mc.signedit.integrations.BreakReplaceSignEditValidator.validate(BreakReplaceSignEditValidator.java:47)
02.04 17:52:02 [Server] INFO at SignEdit.jar//net.deltik.mc.signedit.SignText.applySign(SignText.java:86)
02.04 17:52:02 [Server] INFO at SignEdit.jar//net.deltik.mc.signedit.interactions.SetSignEditInteraction.interact(SetSignEditInteraction.java:50)
02.04 17:52:02 [Server] INFO at SignEdit.jar//net.deltik.mc.signedit.commands.SignCommand.onCommand(SignCommand.java:1133)
02.04 17:52:02 [Server] INFO at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
02.04 17:52:02 [Server] INFO at EssentialsX-2.20.0-dev+43-25ad8ac.jar//com.earth2me.essentials.Essentials.onCommandEssentials(Essentials.java:737)
02.04 17:52:02 [Server] INFO at EssentialsX-2.20.0-dev+43-25ad8ac.jar//com.earth2me.essentials.Essentials.onCommand(Essentials.java:723)
02.04 17:52:02 [Server] INFO at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
02.04 17:52:02 [Server] INFO at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155)
02.04 17:52:02 [Server] INFO at org.bukkit.craftbukkit.v1_19_R2.CraftServer.dispatchCommand(CraftServer.java:929)
02.04 17:52:02 [Server] INFO at org.bukkit.craftbukkit.v1_19_R2.command.BukkitCommandWrapper.run(BukkitCommandWrapper.java:64)
02.04 17:52:02 [Server] INFO at com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:264)
02.04 17:52:02 [Server] INFO at net.minecraft.commands.CommandDispatcher.performCommand(CommandDispatcher.java:313)
02.04 17:52:02 [Server] INFO at net.minecraft.commands.CommandDispatcher.a(CommandDispatcher.java:297)
02.04 17:52:02 [Server] INFO at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:2288)
02.04 17:52:02 [Server] INFO at net.minecraft.server.network.PlayerConnection.lambda$handleChatCommand$20(PlayerConnection.java:2248)
02.04 17:52:02 [Server] INFO at net.minecraft.util.thread.IAsyncTaskHandler.b(IAsyncTaskHandler.java:59)
02.04 17:52:02 [Server] INFO at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
02.04 17:52:02 [Server] INFO at net.minecraft.server.TickTask.run(TickTask.java:18)
02.04 17:52:02 [Server] INFO at net.minecraft.util.thread.IAsyncTaskHandler.d(IAsyncTaskHandler.java:153)
02.04 17:52:02 [Server] INFO at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(IAsyncTaskHandlerReentrant.java:24)
02.04 17:52:02 [Server] INFO at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1341)
02.04 17:52:02 [Server] INFO at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:197)
02.04 17:52:02 [Server] INFO at net.minecraft.util.thread.IAsyncTaskHandler.x(IAsyncTaskHandler.java:126)
02.04 17:52:02 [Server] INFO at net.minecraft.server.MinecraftServer.bh(MinecraftServer.java:1318)
02.04 17:52:02 [Server] INFO at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1311)
02.04 17:52:02 [Server] INFO at net.minecraft.util.thread.IAsyncTaskHandler.c(IAsyncTaskHandler.java:136)
02.04 17:52:02 [Server] INFO at net.minecraft.server.MinecraftServer.i_(MinecraftServer.java:1289)
02.04 17:52:02 [Server] INFO at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1177)
02.04 17:52:02 [Server] INFO at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316)
02.04 17:52:02 [Server] INFO at java.base/java.lang.Thread.run(Thread.java:833)
```

Server version: git-Paper-386 (MC: 1.19.3)
Plugin version: 1.13.7

**Comment by Deltik** [comment]
I was able to reproduce this error by breaking the sign while the edit validation was in progress:

```
[21:32:14 ERROR]: java.lang.ClassCastException: class org.bukkit.craftbukkit.v1_19_R2.block.data.CraftBlockData cannot be cast to class org.bukkit.block.data.type.WallSign (org.bukkit.craftbukkit.v1_19_R2.block.data.CraftBlockData and org.bukkit.block.data.type.WallSign are in unnamed module of loader java.net.URLClassLoader @2530c12)
	at SignEdit.jar//net.deltik.mc.signedit.integrations.BreakReplaceSignEditValidator.getBlockAgainstWallSign(BreakReplaceSignEditValidator.java:94)
	at SignEdit.jar//net.deltik.mc.signedit.integrations.BreakReplaceSignEditValidator.getBlockAgainst(BreakReplaceSignEditValidator.java:86)
	at SignEdit.jar//net.deltik.mc.signedit.integrations.BreakReplaceSignEditValidator.validateBlockPlace(BreakReplaceSignEditValidator.java:71)
	at SignEdit.jar//net.deltik.mc.signedit.integrations.BreakReplaceSignEditValidator.validate(BreakReplaceSignEditValidator.java:47)
	at SignEdit.jar//net.deltik.mc.signedit.SignText.applySign(SignText.java:86)
	at SignEdit.jar//net.deltik.mc.signedit.SignTextHistory.redo(SignTextHistory.java:68)
	at SignEdit.jar//net.deltik.mc.signedit.subcommands.RedoSignSubcommand.execute(RedoSignSubcommand.java:46)
	at SignEdit.jar//net.deltik.mc.signedit.commands.SignCommand.onCommand(SignCommand.java:107)
	at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155)
	at org.bukkit.craftbukkit.v1_19_R2.CraftServer.dispatchCommand(CraftServer.java:929)
	at org.bukkit.craftbukkit.v1_19_R2.command.BukkitCommandWrapper.run(BukkitCommandWrapper.java:64)
	at com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:264)
	at net.minecraft.commands.CommandDispatcher.performCommand(CommandDispatcher.java:313)
	at net.minecraft.commands.CommandDispatcher.a(CommandDispatcher.java:297)
	at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:2288)
	at net.minecraft.server.network.PlayerConnection.lambda$handleChatCommand$20(PlayerConnection.java:2248)
	at net.minecraft.util.thread.IAsyncTaskHandler.b(IAsyncTaskHandler.java:59)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run$$$capture(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java)
	at net.minecraft.server.TickTask.run(TickTask.java:18)
	at net.minecraft.util.thread.IAsyncTaskHandler.d(IAsyncTaskHandler.java:153)
	at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(IAsyncTaskHandlerReentrant.java:24)
	at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1341)
	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:197)
	at net.minecraft.util.thread.IAsyncTaskHandler.x(IAsyncTaskHandler.java:126)
	at net.minecraft.server.MinecraftServer.bh(MinecraftServer.java:1318)
	at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1311)
	at net.minecraft.util.thread.IAsyncTaskHandler.c(IAsyncTaskHandler.java:136)
	at net.minecraft.server.MinecraftServer.i_(MinecraftServer.java:1289)
	at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1177)
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316)
```

I expect this issue to happen fairly rarely under normal circumstances because it is a [race condition](https://en.wikipedia.org/wiki/Race_condition).  If this issue reliably happens every time or frequently when you edit a sign, there is probably another plugin that actually removes or replaces the sign.

I will handle this unhandled error by verifying that the sign is still there during the edit validation, but it will not prevent the sign from being removed by another plugin.  In the case of an undesirable sign removal by another plugin, you will have to figure out what the other plugin is and work with its maintainer to get that bug fixed.  Alternatively, you might be able to avoid the issue by setting `compatibility.edit-validation` to `Standard`, but SignEdit for Bukkit will no longer pretend to break the sign it's editing.

**Comment by Deltik** [comment]
@baailey: Thanks for reporting this bug!

It has now been fixed in [SignEdit for Bukkit v1.13.8](https://github.com/Deltik/SignEdit/releases/tag/v1.13.8), which you can directly download [here](https://github.com/Deltik/SignEdit/releases/download/v1.13.8/SignEdit-1.13.8.jar).
